api/core/workflow/nodes/code/code_node.py

```diff
@@ -118,7 +119,7 @@ def _check_number(self, value: Union[int, float], variable: str) -> Union[int, f
 
         if isinstance(value, float):
             # raise error if precision is too high
-            if len(str(value).split('.')[1]) > MAX_PRECISION:
+            if abs(Decimal(str(value)).as_tuple().exponent) > MAX_PRECISION:
```

Comment:
I think you can remove this `str(...)`.

Reply:
No, the funny thing is, I tried it, and it won't work.
![image](https://github.com/user-attachments/assets/8d1365db-f598-4e85-a49d-55acf6070c5c)